woocommerce/woocommerce
# Break down `create-variable-product` spec into smaller files

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR breaks down the huge `create-variable-product` spec into separate, more focused spec files. This would help in avoiding and troubleshooting flakiness in the future.

These specs are placed inside the new `plugins/woocommerce/tests/e2e-pw/tests/merchant/products/add-variable-product/` folder as an initial step to group related specs together according to the critical flows they test.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Code review

1. Make sure that none of the tests from the [original `create-variable-product`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/tests/merchant/create-variable-product.spec.js) spec got lost during the restructure.
1. Make sure that the new specs do not use the [deprecated `page` methods](https://playwright.dev/docs/api/class-page#deprecated).

#### Test locally

1. Run the setup commands for launching the local environment.
1. `cd` into the `plugins/woocommerce` folder.
1. Run the new specs in the `add-variable-product` folder. They should pass.
   ```bash
   pnpm test:e2e-pw add-variable-product
   ```
1. Run the entire `merchant` E2E tests to see if the new specs conflict with any of the merchant tests.
   ```bash
   pnpm test:e2e-pw tests/e2e-pw/tests/merchant
   ```
#### Test on CI
1. Rerun the E2E tests on CI a few times to make sure no flakiness were added.
